test/cxx/Core/ControllerTest.cpp

```diff
@@ -1144,4 +1144,66 @@ namespace tut {
 		string header = readResponseHeader();
 		ensure(containsSubstring(header, "HTTP/1.1 502"));
 	}
+
+	/***** Application async connection *****/
+
+	TEST_METHOD(60) {
+		set_test_name("Testing async connect flow.");
+		init();
+		mockNextSession();
+		struct ev_async& async = testSession.setupAsync(bg.libev_loop);
```

Comment:
Your strategy with setting up an ev_sync boils down to delaying the accept() call until a moment you choose. But there's really no need to delay the accept call. If you just want to test the async connect flow, then you *only* have to make testSession->initiate() return false, you don't need anything else.

Reply:
I need to delay the accept call so that I can make it take until the connect timeout has run out or the client disconnect has happened.